### ISSUE TITLE
Suggestion: ability to find imports for module

### ISSUE THREAD

**Issue by OliverJAsh** [Microsoft/TypeScript#22374]
Find references is a very useful feature, however sometimes I need to be able to find all imports of a given module:

- regardless of what they're exporting
- but the module does not have any exports

This is sometimes necessary when a module has side effects, and you want to see which files imported it.

This might also be useful for dynamic imports. In one project we have several dynamic imports (for code splitting) that never explicitly reference the module exports, so if I use find references on those module exports, I will not be shown the dynamic import. For example, here is a dynamic import:

![image](https://user-images.githubusercontent.com/921609/37094649-681f42ba-220b-11e8-9ac1-87311528cd16.png)

When I activate find references on the exports of that module, I am not shown the import:

![image](https://user-images.githubusercontent.com/921609/37094636-5bc57f48-220b-11e8-9b72-719a74291ec4.png)

**Comment by mhegazy** [comment]
Find all refs on the module path should show you all imports to it:

![image](https://user-images.githubusercontent.com/8000722/37117550-6f1d56ce-2206-11e8-938f-3b74ec93289f.png)

**Comment by mhegazy** [comment]
Is that not sufficient for your scenarios?

**Comment by OliverJAsh** [comment]
I'm after the opposite—to find all imports for the currently focused module. E.g.:

``` ts
// a.ts
// "find imports" used in this module will list the imports in `b.ts` and `c.ts`
export const foo = 'bar';
```

``` ts
// b.ts
import('./a');
```

``` ts
// c.ts
import './a';
```

**Comment by mhegazy** [comment]
I see.. I suppose we can expose this on the `export` keyword. though it is kinda weird.. possibly another command on the server.. 

@andy-ms, @amcasey and @DanielRosenwasser thoughts?

**Comment by amcasey** [comment]
@OliverJAsh to confirm, you're not looking for consumers of `foo`, you're looking for consumers of `a.ts`?

**Comment by amcasey** [comment]
@mhegazy I agree that it's not clear how we would trigger such a search.  Personally, I would expect invoking Find All References on the `export` keyword to list references to *that* exported symbol.

**Comment by mhegazy** [comment]
> Personally, I would expect invoking Find All References on the export keyword to list references to that exported symbol.

yeah.. then a new command is needed, but it wounder if that will be discoverable.
